provider/designate/designate.go

```diff
@@ -216,13 +218,13 @@ func (c designateClient) CreateRecordSet(zoneID string, opts recordsets.CreateOp
 
 // UpdateRecordSet updates recordset in the given DNS zone
 func (c designateClient) UpdateRecordSet(zoneID, recordSetID string, opts recordsets.UpdateOpts) error {
-	_, err := recordsets.Update(c.serviceClient, zoneID, recordSetID, opts).Extract()
+	_, err := recordsets.Update(context.TODO(), c.serviceClient, zoneID, recordSetID, opts).Extract()
 	return err
 }
 
 // DeleteRecordSet deletes recordset in the given DNS zone
 func (c designateClient) DeleteRecordSet(zoneID, recordSetID string) error {
```

Comment:
Probaly a cleaner appraoch is to add context to methods 

example
```go
DeleteRecordSet(ctx Context,...)
```

Reply:
Hi @ivankatliarchuk , thanks for the review.
We are not experts of this component and we thought we'd support you in the bump to Gophercloud v2 (with v1 in code freeze).
Adding these temporary `context.TODO()` lets us keep identical behaviour in the component across the bump, and enables you to wire `context.Context` throughout the codebase whenever you are ready to do that. What would you think if we just leave the `context.TODO()` calls in, and you consider adding `context.Context` to the methods and their interfaces at a later stage?

Reply:
We will get back to this discussion when this is resolved https://github.com/kubernetes-sigs/external-dns/pull/5021#issuecomment-2662908611 if you don't mind

Reply:
The release cycle is 6 months at the moment. If something is broken, we get the  provider fixed in a year time ;-)

@mloiseleur wdyt?